packages/manager/modules/backup-agent/src/utils/__tests__/countBackupAgents.test.ts

```diff
@@ -0,0 +1,44 @@
+import { describe, expect, it } from 'vitest';
+import { countBackupAgents } from '@/utils/countBackupAgents';
+import type { VSPCTenant } from '@/types/VspcTenant.type';
+
+describe('countBackupAgents test suite', () => {
+  it('returns 0 when tenants array is empty', () => {
+    expect(countBackupAgents([])).toBe(0);
+  });
+
+  it('returns 0 when tenants have no currentState', () => {
+    const tenants: VSPCTenant[] = [
+      { id: '1', backupAgents: undefined } as any
+    ];
+    expect(countBackupAgents(tenants)).toBe(0);
+  });
+
+  it('returns 0 when backupAgents is undefined or null', () => {
+    const tenants: VSPCTenant[] = [
+      { id: '1', backupAgents: undefined } as any,
+      { id: '2', backupAgents: null } as any
+    ];
+    expect(countBackupAgents(tenants)).toBe(0);
+  });
```

Comment:
If your function support undefined and null. You can adapt your params type instead add `as any` :)